mirrord/layer/src/file/ops.rs

```diff
@@ -217,6 +219,91 @@ pub(crate) fn fdopen(fd: RawFd, rawish_mode: Option<&CStr>) -> Detour<*mut FILE>
     Detour::Success(result)
 }
 
+/// creates a directory stream for the `remote_fd` in the agent
+#[tracing::instrument(level = "trace")]
+pub(crate) fn fdopendir(fd: RawFd) -> Detour<usize> {
+    // usize == ptr size
+    // we don't return a pointer to an address that contains DIR
+
+    let open_files = OPEN_FILES.lock()?;
+    let remote_file_fd = open_files.get(&fd).ok_or(Bypass::LocalFdNotFound(fd))?;
+
+    let (dir_channel_tx, dir_channel_rx) = oneshot::channel();
+
+    let open_dir_request = OpenDir {
+        remote_fd: *remote_file_fd,
+        dir_channel_tx,
+    };
+
+    blocking_send_file_message(HookMessageFile::OpenDir(open_dir_request))?;
+
+    let OpenDirResponse { fd } = dir_channel_rx.blocking_recv()??;
+
+    let fake_local_dir_name = CString::new(fd.to_string())?;
```

Comment:
Maybe add some comments here about why there's a fake dir name, how it will be used and by who?

Reply:
This is a common pattern in code, we can improve it later if we want (I'd change the signature of fake file to just use remote fd and generate the fake name there as all code repeats it)